src/components/modules/blockManager.ts

```diff
@@ -647,18 +681,65 @@ export default class BlockManager extends Module {
   }
 
   /**
-   * Bind Events
+   * Bind Block events
    *
    * @param {Block} block - Block to which event should be bound
    */
-  private bindEvents(block: Block): void {
+  private bindBlockEvents(block: Block): void {
     const { BlockEvents, Listeners } = this.Editor;
 
-    Listeners.on(block.holder, 'keydown', (event) => BlockEvents.keydown(event as KeyboardEvent), true);
-    Listeners.on(block.holder, 'mousedown', (event: MouseEvent) => BlockEvents.mouseDown(event));
-    Listeners.on(block.holder, 'keyup', (event) => BlockEvents.keyup(event));
-    Listeners.on(block.holder, 'dragover', (event) => BlockEvents.dragOver(event as DragEvent));
-    Listeners.on(block.holder, 'dragleave', (event) => BlockEvents.dragLeave(event as DragEvent));
+    this.listenerIds.push(
+      Listeners.on(block.holder, 'keydown', (event: KeyboardEvent) => {
+        BlockEvents.keydown(event);
+      }, true)
+    );
+
+    this.listenerIds.push(
+      Listeners.on(block.holder, 'keyup', (event: KeyboardEvent) => {
+        BlockEvents.keyup(event);
+      })
+    );
+
+    this.listenerIds.push(
+      Listeners.on(block.holder, 'dragover', (event: DragEvent) => {
+        BlockEvents.dragOver(event);
+      })
+    );
+
+    this.listenerIds.push(
+      Listeners.on(block.holder, 'dragleave', (event: DragEvent) => {
+        BlockEvents.dragLeave(event);
+      })
+    );
+  }
+
+  /**
+   * Disable all handlers and bindings
+   * The sequence is following:
+   *  - Removes all listeners by id
+   *  - Removes all shortcuts
+   */
+  private disableModuleBindings(): void {
+    const { Listeners } = this.Editor;
+
+    for (const id of this.listenerIds) {
+      Listeners.offById(id);
+    }
+
+    this.listenerIds = [];
+  }
+
+  /**
+   * Enables all module handlers and bindings
+   * The sequence is following:
+   *  - Enable shortcuts again
+   *  - Restore `copy` and `cut` bindings
+   *  - Bind all events handlers for all Blocks
+   */
+  private enableModuleBindings(): void {
+    this.blocks.forEach((block: Block) => {
+      this.bindBlockEvents(block);
+    });
```

Comment:
You can make `bindBlockEvents` arrow function and write here like this:

```suggestion
    this.blocks.forEach(this.bindBlockEvents);
```

Reply:
what is the difference?

Reply:
It is not good to make some methods as arrows function and some not don't you think so? we need to have an agreement that if it is a class method so they won't be arrow function, but for lambdas or nested functions it is very useful to use them instead of `.bind()`

Reply:
Writing class methods that are event handlers or used in map array methods as arrow functions is a common practice.
The advantages are important: 
1. 
```
Listeners.on(element, this.bindEvent) //more readable than:
Listeners.on(element, (e) => {
  this.bindEvent(e)
});
```

2. If you pass an arrow function as event handler in-place, you can't remove this handler as function is anonymous.
Moreover, if for some reason, `addEventListener` is called twice (or more), there will be 2 same handlers on one event. Whereas, if you pass named arrow function (or arrow class method), user agent won't add two listeners and you would be able to remove handler by passing the same function object to `removeEventListener`.,